homeassistant/components/yessssms/notify.py

```diff
@@ -8,34 +8,57 @@
 
 from homeassistant.components.notify import PLATFORM_SCHEMA, BaseNotificationService
 
+from YesssSMS import YesssSMS
+
+from .const import CONF_PROVIDER, CONF_TEST_LOGIN_DATA
+
 _LOGGER = logging.getLogger(__name__)
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Required(CONF_USERNAME): cv.string,
         vol.Required(CONF_PASSWORD): cv.string,
         vol.Required(CONF_RECIPIENT): cv.string,
+        vol.Optional(CONF_PROVIDER, default="YESSS"): cv.string,
+        vol.Optional(CONF_TEST_LOGIN_DATA, default=True): cv.boolean,
```

Comment:
Please remove the option.

Reply:
Why? it's on by default.
And it allows to stop checking working credentials. I don't see why advanced users should not turn this off to spare providers from a test login on every startup.

Reply:
We don't have this option in any integration since it's standard to check credentials.

If this integration is later reworked to use config entries this check can be avoided on restart.

Reply:
ok I could live with the default credentials check. but...

What should be done if the login credentials do not work? disable the notify component (aka return None) or go along and just log the failed credential check?

consider this scenario:
you have your alarm system running with HA. your node reboots, for some reason. it's NOT connected to the internet at startup. 
Why should working credentials not be accepted because of a failed check at startup? they should be accepted. because if the node is online later on, sending SMS will work.

Reply:
which raises another issue. I need to catch more connection issue Exceptions upstream.

Reply:
The standard is to disable the integration on bad credentials. The integration should differentiate between bad credentials and bad connection.

Reply:
YES. ok. that's also my take.

Disable the integration when validation of credentials fails.
but silently accept them if offline.

the error handling would let you know that credentials are bad anyhow. But the afore mentioned scenario would work.

Reply:
If we can we should retry the set up process if connection is bad. I'm not sure we have that option for notify platforms. We do have it when setting up the config entry. I can check this and get back to you.

Regular entity platforms can raise `PlatformNotReady`.

Reply:
I removed the CONF_TEST_LOGIN_DATA option.

Also I'm catching connection errors on startup. The mentioned scenario should be fine now - as login issues will be logged later.

Reply:
hahahaha!
And now yesss.at is down and provides the perfect testing ground for this scenario :smile_cat:. 
HA should start with unchecked credentials... SMS sending would be possible when they are back.